src/core/utils.ts

```diff
@@ -28,3 +29,10 @@ export function mapToSearchActions(values: string[], actions: typeof FluxCapacit
 }
 
 export const rayify = <T>(values: T | T[]): T[] => Array.isArray(values) ? values : [values];
+
+export const arrayToDotNotation = (input: string): string => {
+  if (typeof input !== 'string') {
+    throw new Error('input not a string');
+  }
+  return input.replace(/\[([^\]]+)\]/g, '.$1');
```

Comment:
we only want to match numeric values between the brackets so this regex can be updated to 
`/\[([\d.]+)\]/g`

Reply:
Also, to avoid re-compiling the regex every time `arrayToDotNotation()` is called, we should save it as a `const` at the top of this file

Reply:
Oh neat, I didn't know that there was additional overhead with having to re-compile regex patterns if not saved in a const.

I used \[(\d+)\] over \[([\d.]+)\] since I don't think we need to consider a[1.2]?